### PR TITLE
Dev/proullon/auth

### PR DESCRIPTION
**Pull request by proullon**
!! contains breaking change

### CODE REVIEW

File: engine/hatchery/main.go [hunk 1]
```diff
@@ -225,24 +226,23 @@ func parseConfig(cmd *cobra.Command) HatcheryMode {
 		sdk.Exit("CDS api endpoint not provided. See usage:\n%s\n", cmd.Short)
 	}
 
-	usr := viper.GetString("cds-user")
-	passwd := viper.GetString("cds-password")
+	uk = viper.GetString("token")
+
 	token := os.Getenv("CDS_TOKEN")
```
Comment (fsamin) [inline engine/hatchery/main.go]: uk = viper.GetString("token")
token := os.Getenv("CDS_TOKEN")

one of those could be removed

File: engine/api/worker/worker.go [hunk 2]
```diff
@@ -239,13 +240,25 @@ func RegisterWorker(db *sql.DB, name string, uk string, modelID int64, hatcheryI
 		return nil, fmt.Errorf("cannot register worker with empty worker key")
 	}
 
-	// Load user key
-	userID, e, err := LoadUserKey(db, uk)
+	/// Load token
+	var userID, groupID int64
+	var e sdk.Expiration
+	t, err := LoadToken(db, uk)
+	// /!\ LEGACY:  Load user key
```
Comment (fsamin) [inline engine/api/worker/worker.go]: I suggest to not keep legacy code.

File: engine/api/build.go [hunk 3]
```diff
@@ -554,7 +554,14 @@ func getQueueHandler(w http.ResponseWriter, r *http.Request, db *sql.DB, c *cont
 		}
 	}
 
-	queue, err := build.LoadUserWaitingQueue(db, c.User)
+	var queue []sdk.ActionBuild
+	var err error
+	// LEGACY
```
Comment (fsamin) [inline engine/api/build.go]: Le's remove legacy code

File: sdk/worker.go [hunk 4]
```diff
@@ -8,14 +8,16 @@ import (
...
-	Owner      User      `json:"owner"`
-	Model      int64     `json:"model"`
-	HatcheryID int64     `json:"hatchery_id"`
-	Status     Status    `json:"status"` // Waiting, Building, Disabled, Unknown
+	ID       string    `json:"id"`
+	Name     string    `json:"name"`
+	LastBeat time.Time `json:"-"`
+	GroupID  int64     `json:"group_id"`
+	// Owner is deprecated in favor of Group for permission
+	OwnerID    int64  `json:"-"`
```
Comment (fsamin) [inline sdk/worker.go]: Remove the OwnerID and all the things...

File: engine/api/main.go [hunk 5]
```diff
@@ -284,7 +290,7 @@ func (router *Router) init() {
 	router.Handle("/project/{key}/application/{permApplicationName}/pipeline/{permPipelineKey}/history", GET(getPipelineHistoryHandler))
 	router.Handle("/project/{key}/application/{permApplicationName}/pipeline/{permPipelineKey}/build/{build}/log", GET(getBuildLogsHandler))
 	router.Handle("/project/{key}/application/{permApplicationName}/pipeline/{permPipelineKey}/build/{build}/test", POST(addBuildTestResultsHandler), GET(getBuildTestResultsHandler))
-	router.Handle("/project/{key}/application/{permApplicationName}/pipeline/{permPipelineKey}/build/{build}/variable", POST(addBuildVariableHandler))
+	router.Handle("/project/{key}/application/{app}/pipeline/{permPipelineKey}/build/{build}/variable", POSTEXECUTE(addBuildVariableHandler))
```
Comment (sguiheux) [inline engine/api/main.go]: strange, worker are running pipeline on an application. He must have execution right on this application

Comment (proullon) [inline engine/api/main.go]: Actually no the worker starts and build without app permissions, variable export is the only feature not working without app permission so it's super counter-intuitive

**Pull request merged**